pkg/server/image_pull.go

```diff
@@ -280,68 +262,63 @@ func (c *criService) getTLSConfig(registryTLSConfig config.TLSConfig) (*tls.Conf
 	return tlsConfig, nil
 }
 
-// getResolver tries registry mirrors and the default registry, and returns the resolver and descriptor
-// from the first working registry.
-func (c *criService) getResolver(ctx context.Context, ref string, cred func(string) (string, string, error)) (remotes.Resolver, imagespec.Descriptor, error) {
-	refspec, err := reference.Parse(ref)
-	if err != nil {
-		return nil, imagespec.Descriptor{}, errors.Wrap(err, "parse image reference")
-	}
+// registryHosts is the registry hosts to be used by the resolver.
+func (c *criService) registryHosts(auth *runtime.AuthConfig) docker.RegistryHosts {
+	return func(host string) ([]docker.RegistryHost, error) {
+		var registries []docker.RegistryHost
 
-	var (
-		transport  = newTransport()
-		httpClient = &http.Client{Transport: transport}
-	)
-
-	// Try mirrors in order first, and then try default host name.
-	for _, e := range c.config.Registry.Mirrors[refspec.Hostname()].Endpoints {
-		u, err := url.Parse(e)
+		// Try mirrors in order, and then try the default registry if not tried.
+		defaultHost, err := docker.DefaultHost(host)
 		if err != nil {
-			return nil, imagespec.Descriptor{}, errors.Wrapf(err, "parse registry endpoint %q", e)
+			return nil, errors.Wrapf(err, "get default host")
+		}
+		defaultEndpoint := "https://" + defaultHost
```

Comment:
docker registry can also communicate without ssl, right? ...do we need a todo for this? there may be an issue already open for non https defaults I think I saw one a while back.

Reply:
If users want insecure access, they can configure mirror `http://xxxx` to overwrite the default.

The default should be `https` :)

Reply:
Ok but if I remember right the issue opened by a customer was failing with an https error.. when setting a non http default.   hmm  maybe filter out https:/default if they included a mirror to http:/default...?

Reply:
@mikebrow You are right. I'll fix it.

Reply:
Done